app/components/MemberForm/MemberForm.tsx

```diff
@@ -61,175 +68,209 @@ const resolver = createResolver<FormSchema>({
   gender: memberSchema.gender(),
   nationality: memberSchema.nationality(),
   religion: memberSchema.religion(),
+  dob: memberSchema.dob(),
 });
 
-// eslint-disable-next-line max-lines-per-function
-export const MemberForm = () => {
-  const [opened, { toggle }] = useDisclosure(true);
+const t = messages.member;
+const tt = t.createForm;
+
+interface Props {
+  initialMember?: Member;
+  familyId: string;
+}
 
-  const tt = messages.member;
-  const t = tt.createForm;
+export const MemberForm = ({ initialMember, familyId }: Props) => {
+  const [createMemberMutation] = useCreateMemberMutation();
+  const [opened, { toggle }] = useDisclosure(true);
+  const [isEditableMode, { toggle: toggleEditableMode }] = useToggle(
+    !initialMember,
+  );
```

Comment:
does the existence of an initial member imply it's in editable mode?
why not rely in status?
```suggestion
  const [isEditableMode, { toggle: toggleEditableMode }] = useToggle(
    initialMember?.status === 'completed',
  );
```